step-api/src/main/java/org/jenkinsci/plugins/workflow/steps/AbstractSynchronousNonBlockingStepExecution.java

```diff
@@ -0,0 +1,84 @@
+package org.jenkinsci.plugins.workflow.steps;
+
+import hudson.util.DaemonThreadFactory;
+import hudson.util.NamingThreadFactory;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+/**
+ * Similar to {@link AbstractSynchronousStepExecution} but it always executes synchronously and does not block the CPS VM thread.
```

Comment:
> Similar to `AbstractSynchronousStepExecution` but it always executes synchronously

does not make any sense: `AbstractSynchronousStepExecution` always executes synchronously too, of course.